src/schematics/setup/index.ts

```diff
@@ -33,7 +33,9 @@ export const setupProject =
 
     const { project, projectName } = getProject(config, tree);
 
-    const sourcePath = [project.root, project.sourceRoot].filter(it => !!it).join('/');
+    const sourcePath = project.sourceRoot?.startsWith(project.root)
+      ? project.sourceRoot
+      : [project.root, project.sourceRoot].filter(it => !!it).join('/');
```

Comment:
Now I'm doubting myself, wonder if this logic is wrong entirely... perhaps all this should be `project.sourceRoot || project.root` instead if they're both relative. WDYT?

Reply:
In Nx land I can confirm that we always treat both of these as workspace root, but I'm not sure if thats the case for all regular angular workspaces. I was trying to keep this from being a "breaking" fix at all, I think just `project.sourceRoot ?? project.root` would be much simpler, if you are good with that fix.

Reply:
Lets go for that. I think in coding this I misunderstood the paths, I was pretty rushed to get the schematics out the door.

Reply:
Sure, I just pushed up those changes.